### ISSUE TITLE
[Feature] - Ability to Resize Volumes Through PVC

### ISSUE THREAD

**Issue by samson4649** [vultr/vultr-csi#90]
**Is your feature request related to a problem? Please describe.**
Currently the Vultr CSI doesnt allows resizing PVCs as needed. It would be beneficial to allow PVCs to be expanded as required to help customers manage costs.

**Describe the solution you'd like**
Vultr CSI processes requests to change provisioned volume based on the `allowVolumeExpansion` storage class option. 

```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: vultr-block-storage-resizable
parameters:
  block_type: high_perf
provisioner: block.csi.vultr.com
reclaimPolicy: Delete
volumeBindingMode: Immediate
allowVolumeExpansion: true  # <--- here
```

**Describe alternatives you've considered**
The only  way around this is to use a storage class backed by another CSI either external to Vultr or deployed on other Vultr infra.

**Extra information**
Below is the current output when creating another storage class using the Vultr CSI with `allowVolumeExpansion` enabled then editing the PVC for expansion.

```
Events:
  Type     Reason                 Age                From                                                                             Message
  ----     ------                 ----               ----                                                                             -------
  Normal   Provisioning           19m                block.csi.vultr.com_csi-vultr-controller-0_8864b530-9d56-4b1e-8f66-95f458a43e56  External provisioner is provisioning volume for claim "testing-pvc/testing-pvc"
  Normal   ProvisioningSucceeded  19m                block.csi.vultr.com_csi-vultr-controller-0_8864b530-9d56-4b1e-8f66-95f458a43e56  Successfully provisioned volume pvc-0761d6f94e434df6
  Warning  ExternalExpanding      4m37s              volume_expand                                                                    Ignoring the PVC: didn't find a plugin capable of expanding the volume; waiting for an external controller to process this PVC.
```

**Comment by optik-aper** [comment]
Thanks for the request.  We'll look into this and let you know.

**Comment by ddymko** [comment]
@optik-aper @samson4649 

This was never implemented because Vultr block storage doesn't currently support a live expansion. It requires a restart of the instance it's attached to and some minor adjustments. 

The goal would be to have Vultr block offer live expansion without user intervention to make this a more seamless process for the CSI/K8

**Comment by optik-aper** [comment]
@ddymko Thanks for the context.

**Comment by koenbeuk** [comment]
Are there guides on how to expand a block storage volume in kubernetes? I'm new to Kubernetes, followed the docs and got a 10gb PVC that hosts a database. I now run out of storage space and need to expand this volume. Downtime is acceptable. 

I tried updating the storage quota and indeed that fails with the message: `only dynamically provisioned pvc can be resized and the storageclass that provisions the pvc must support resize`

I can look into the workarounds as offered by @samson4649 but that seems like work... Could I simply resize the volume through the Vultr portal and restart my database server? Thanks!

**Comment by gooooer** [comment]
Hi @ddymko ! Has there been any progress since then? Would it be possible to resize Vultr's backed volumes per k8s's manual here https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/ ?

**Comment by optik-aper** [comment]
@gooooer It's not supported yet but the team has it on their roadmap.  We'll make sure to support it when the time comes.

**Comment by happytreees** [comment]
@samson4649 This feature has been implemented into the CSI as of v0.9.0 #116
